ui/src/router/edit/EditRouterView.js

```diff
@@ -57,13 +108,17 @@ const EditRouterView = ({ projectId, currentRouter, ...props }) => {
         ) : (
           <VersionComparisonView
             currentRouter={currentRouter}
-            updatedRouter={updatedRouter}
+            updatedRouter={routerConfig}
             onPrevious={toggleDiffView}
             onSubmit={onSubmit}
-            isSubmitting={submissionResponse.isLoading}
+            isSubmitting={
+              updateRouterResponse.isLoading ||
+              createRouterVersionResponse.isLoading
+            }
+            setWithDeployment={setWithDeployment}
```

Comment:
Minor suggestion: I think rather than `setWithDeployment` prop (which sounds like a detail internal to the `EditRouterView` component, we can create 2 onsubmit props on the `VersionComparisonView` like:

```javascript
<VersionComparisonView
...
onSave={() => {setWithDeployment(false); onSubmit();}
onDeploy={() => {setWithDeployment(true); onSubmit();}
```